OJS.Workers.ExecutionStrategies/RunSpaAndExecuteMochaTestsExecutionStrategy.cs

```diff
@@ -0,0 +1,389 @@
+            var allowedFileExtensions = (!trimmedAllowedFileExtensions?.StartsWith(".") ?? false)
+                ? $".{trimmedAllowedFileExtensions}"
+                : trimmedAllowedFileExtensions;
+
+            if (allowedFileExtensions != Constants.ZipFileExtension)
+            {
+                throw new ArgumentException("Submission file is not a zip file!");
+            }
+        }
+
+        private ICollection<TestResult> ExtractTestResultsFromReceivedOutput(string receivedOutput, int parentTestId)
+        {
+            JsonExecutionResult mochaResult = JsonExecutionResult.Parse(this.PreproccessReceivedExecutionOutput(receivedOutput));
+            if (mochaResult.TotalTests == 0)
+            {
+                return new List<TestResult>
+                {
+                    new TestResult
+                    {
+                        Id = parentTestId,
+                        IsTrialTest = false,
+                        ResultType = TestRunResultType.WrongAnswer,
+                        CheckerDetails = new CheckerDetails
+                        {
+                            UserOutputFragment = receivedOutput
+                        }
+                    }
+                };
+            }
+
+            return mochaResult.TestErrors
+                .Select(test => this.ParseTestResult(test, parentTestId))
+                .ToList();
+        }
+
+        private TestResult ParseTestResult(string testResult, int parentTestId)
+            => new TestResult
+            {
+                Id = parentTestId,
+                IsTrialTest = false,
+                ResultType = testResult == null
+                    ? TestRunResultType.CorrectAnswer
+                    : TestRunResultType.WrongAnswer,
+                CheckerDetails = testResult == null
+                    ? default
+                    : new CheckerDetails {UserOutputFragment = testResult},
+            };
+
+        private string PreproccessReceivedExecutionOutput(string receivedOutput)
+            => Regex.Unescape(receivedOutput)
+                .Trim()
+                .Replace("b'", string.Empty)
+                .Replace("}'", "}")
+                .Replace("}None", "}");
+
+        private string SavePythonCodeTemplateToTempFile()
+        {
+            var pythonCodeTemplate = this.PythonCodeTemplate.Replace("\\", "\\\\");
+            return FileHelpers.SaveStringToTempFile(this.WorkingDirectory, pythonCodeTemplate);
+        }
+
+        private void SaveNginxFile()
+            => FileHelpers.SaveStringToFile(this.NginxFileContent, this.NginxConfFileFullPath);
+
+        private void SaveTestsToFiles(IEnumerable<TestContext> tests)
+        {
+            foreach (var test in tests)
+            {
+                var testInputContent = test.Input
+                    .Replace(UserApplicationHttpPortPlaceholder, this.PortNumber.ToString());
+                    // .Replace("localhost", "host.docker.internal");
```

Comment:
Why this is left commented?

Reply:
Not sure, maybe something Doncho has changed